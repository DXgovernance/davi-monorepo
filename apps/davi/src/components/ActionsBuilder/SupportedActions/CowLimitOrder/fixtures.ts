import { DecodedCall, SupportedAction } from 'components/ActionsBuilder/types';
import { BigNumber, utils } from 'ethers';
import { MOCK_ADDRESS } from 'hooks/Guilds/ens/fixtures';
import CowSettlement from 'contracts/CowSettlement.json';

const CowSettlementContract = new utils.Interface(CowSettlement.abi);

export const limitOrderDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.COW_SWAP_LIMIT_ORDER,
  function: CowSettlementContract.getFunction('setPreSignature'),
  to: '0xfFb1cd0F95368DDd06D556161c5D3d9f0f4Fe6d2',
  value: BigNumber.from(0),
  args: {
    orderUid: '',
    signed: '',
  },
  optionalProps: {
    sellAmount: BigNumber.from('5000000000000000000'),
    buyToken: MOCK_ADDRESS,
    sellToken: MOCK_ADDRESS,
    functionSignature: '',
  },
};
