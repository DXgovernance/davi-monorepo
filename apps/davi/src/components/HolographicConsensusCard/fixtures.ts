import { BigNumber } from 'ethers';
import { IStakes } from './types';

export const mockProposalStakeDetails: IStakes = {
  for: [
    {
      id: '1',
      staker: '0xaf8eb8c3a5d9d900aa0b98e3df0bcc17d3c5f698',
      amount: BigNumber.from('100000000000000000000'),
    },
    {
      id: '2',
      staker: '0xc5b20ade9c9cd5e0cc087c62b26b815a4bc1881f',
      amount: BigNumber.from('100000000000000000000'),
    },
    {
      id: '3',
      staker: '0xc5b20ade9c9cd5e0cc087c62b26b815a4bc1881f',
      amount: BigNumber.from('336000000000000000000'),
    },
    {
      id: '4',
      staker: '0xc5b20ade9c9cd5e0cc087c62b26b815a4bc1881f',
      amount: BigNumber.from('236600000000000000000'),
    },
    {
      id: '5',
      staker: '0xc5b20ade9c9cd5e0cc087c62b26b815a4bc1881f',
      amount: BigNumber.from('47138000000000000000'),
    },
  ],
  against: [
    {
      id: '6',
      staker: '0x84eeb305da0a4309a696d43de9f79f04e66eb4f8',
      amount: BigNumber.from('100000000000000000000'),
    },
  ],
};

export const mockProposalStakeDetailsEmpty = {
  for: [],
  against: [],
};
