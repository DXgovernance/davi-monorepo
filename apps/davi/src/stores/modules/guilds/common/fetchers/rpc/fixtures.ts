export const MOCK_CONTRACT_ADDRESS =
  '0x0000000000000000000000000000000000000001';
export const MOCK_GUILD_ADDRESS = '0x0000000000000000000000000000000000000002';
