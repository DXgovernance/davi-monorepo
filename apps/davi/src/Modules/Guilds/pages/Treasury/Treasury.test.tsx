import { render } from 'utils/tests.js';
import { Treasury } from '.';
import { MOCK_TOKEN_INFO } from './fixtures';

jest.mock('Modules/Guilds/Hooks/useTypedParams', () => ({
  useTypedParams: () => ({
    chainName: 'localhost',
    guildId: '0xE9bDaB08f2FBb370d2a6F6661a92d9B6157E9fd2',
  }),
}));

jest.mock('hooks/Guilds/erc20/useAllERC20Balances', () => ({
  useAllERC20Balances: () => ({
    data: [MOCK_TOKEN_INFO],
    isLoading: false,
    isError: false,
  }),
}));

describe('Treasury Page', () => {
  it(`Should render without data`, async () => {
    const { container } = render(<Treasury />);
    expect(container).toMatchSnapshot();
  });
});
