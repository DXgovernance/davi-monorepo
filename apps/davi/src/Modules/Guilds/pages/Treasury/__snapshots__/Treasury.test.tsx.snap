// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Treasury Page Should render without data 1`] = `
.c4 {
  width: 100%;
  min-width: 200px;
  height: 1px;
  background-color: #303338;
  margin: 0;
  padding: 0;
}

.c0 {
  box-sizing: 'border-box';
  min-width: 0;
  margin: 0;
  padding: 0;
}

.c2 {
  font-family: 'Inter',-apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
}

.c2 strong,
.c2 b {
  font-weight: 600;
}

.c1 {
  border-radius: 10px;
  background-color: #181B1B;
  padding: 1px 24px 8px 24px;
}

.c3 {
  color: #DEFF4E;
}

.c5 {
  margin: 0;
  margin-left: -24px;
  width: calc(100% + 48px);
}

.c6 {
  width: 100%;
  border-collapse: collapse;
  margin-top: 1rem;
}

.c7 {
  border-bottom: 1px solid #2F3533;
}

.c8 {
  padding-bottom: 12px;
  font-weight: 400;
  text-align: left;
  font-size: 12px;
  color: #A1A6B0;
}

.c9 {
  padding-bottom: 12px;
  font-weight: 400;
  text-align: right;
  font-size: 12px;
  color: #A1A6B0;
}

<div>
   
  <div
    class="c0 c1"
  >
    <h2
      class="c2 c3"
      size="1"
    >
      treasury.treasury
    </h2>
    <div
      class="c4 c5"
    />
    <table
      class="c6"
    >
      <thead
        class="c7"
      >
        <tr>
          <th
            class="c8"
          >
            treasury.asset
          </th>
          <th
            class="c9"
          >
             treasury.tokenAmount
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
   
</div>
`;
